botocove/cove_host_account.py

```diff
@@ -128,128 +118,98 @@ def _get_boto3_sts_client(self, assuming_session: Optional[Session]) -> STSClien
+        )
 
-        account_list: List[str] = []
+    def _is_organizational_unit_id(self, resource_id: str) -> bool:
+        return bool(re.match(r"^ou-[0-9a-z]{4,32}-[a-z0-9]{8,32}$", resource_id))
 
-        for parent_ou in target_ous:
+    def _is_account_id(self, resource_id: str) -> bool:
+        return bool(re.match(r"^\d{12}$", resource_id))
 
-            current_ou_list: List[str] = []
+    def _list_descendant_accounts(self, ancestor_id: str) -> Set[str]:
+        descendant_accounts = self._list_accounts_for_parent(ancestor_id)
 
-            # current_ou_list is mutated and recursivly populated with all childs
-            self._get_all_child_ous(parent_ou, current_ou_list)
+        child_ous = self._list_organizational_units_for_parent(ancestor_id)
 
-            # for complete list add parent ou as well to list of child ous
-            current_ou_list.append(parent_ou)
+        for ou in child_ous:
+            descendant_accounts.update(self._list_descendant_accounts(ou))
 
-            account_list.extend(
-                self._get_accounts_by_organization_units(current_ou_list)
-            )
+        return descendant_accounts
 
-        return account_list
+    @lru_cache
+    def _list_accounts_for_parent(self, parent_id: str) -> Set[str]:
+        pages = self.org_client.get_paginator("list_accounts_for_parent").paginate(
+            ParentId=parent_id
+        )
 
-    def _get_all_child_ous(self, parent_ou: str, ou_list: List[str]) -> None:
-        """Depth-first recursion mutates the current_ou_list present in the calling
-        function to establish all children of a parent OU"""
-        child_ous = self._get_child_ous(parent_ou)
-        child_ous_list = [ou["Id"] for ou in child_ous["Children"]]
-        ou_list.extend(child_ous_list)
+        accounts = {account["Id"] for page in pages for account in page["Accounts"]}
 
-        for ou in child_ous_list:
-            self._get_all_child_ous(ou, ou_list)
+        return accounts
 
-    def _get_accounts_by_organization_units(
-        self, organization_units: List[str]
-    ) -> List[str]:
+    @lru_cache
+    def _list_organizational_units_for_parent(self, parent_id: str) -> Set[str]:
+        pages = self.org_client.get_paginator(
+            "list_organizational_units_for_parent"
+        ).paginate(ParentId=parent_id)
 
-        account_list: List[str] = []
+        organizational_units = {
+            organizational_unit["Id"]
+            for page in pages
+            for organizational_unit in page["OrganizationalUnits"]
+        }
 
-        for ou in organization_units:
-            ou_children = self._get_child_accounts(ou)
-            account_list.extend(acc["Id"] for acc in ou_children["Children"])
+        return organizational_units
 
-        return account_list
+    def _map_active_accounts(self) -> Dict[str, AccountTypeDef]:
+        pages = self.org_client.get_paginator("list_accounts").paginate()
```

Comment:
I think (only had a quick pass!) this breaks functionality for non-organizations users - previously passing a list of IDs when running from any account with cross account trust to the list would run. This is particularly important for friends in the more esoteric AWS regions without AWS Orgs :)

Reply:
You're surely right. I've never used botocove in that setup. Even if I pass a list of account IDs, I'm always using it in an organization.

Is it the `org_master=False` option that make it behave in non-org mode?

It would help to have some unit tests for that mode. Can we simulate it with moto?

Reply:
I think with my latest commit this would be supported now. If you think it would help we could add a test that uses `org_master=False` and passes some `target_ids` to prove it.